scene/util.py

```diff
@@ -0,0 +1,13 @@
+def apply_halo_rotation(scene_object, ijkw):
```

Comment:
Everywhere you use this you explicitly pass in a tuple that unpacks an object's `rot_i`, `rot_j`, etc... values. That's duplicated logic you need to remember every time you want to use this function. I'd recommend passing the object itself in and have this function pull the `rot_i` etc... itself. 

If you want the improved flexibility of specifying each component yourself, you might also consider supporting `kwargs` for the quaternion components.

Reply:
I don't want to limit this function to only working with Jms classes. kwargs are workable though.

Reply:
You could "overload" it to take a `jms` kwarg. That seems a little smelly to me, but it's an option.